File: ObjCSample/SampleApp/ViewController.m
#import "ViewController.h"

#import "CodeNextViewController.h"
#import "StoryboardNextViewController.h"

@implementation ViewController

- (IBAction)showAlert
{
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Title"
                                                                             message:@"Message"
                                                                      preferredStyle:UIAlertControllerStyleAlert];
    [self setUpActionsForAlertController:alertController];
    [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField) {
        textField.placeholder = @"Placeholder";
    }];
    [self presentViewController:alertController animated:YES completion:self.alertPresentedCompletion];
}

- (IBAction)showActionSheet:(id)sender
{
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Title"
                                                                             message:@"Message"
                                                                      preferredStyle:UIAlertControllerStyleActionSheet];
    [self setUpActionsForAlertController:alertController];

    UIPopoverPresentationController *popover = alertController.popoverPresentationController;
    if (popover)
    {
        popover.sourceView = sender;
        popover.sourceRect = [sender bounds];
        popover.permittedArrowDirections = UIPopoverArrowDirectionAny;
    }

    [self presentViewController:alertController animated:YES completion:NULL];
}

- (void)setUpActionsForAlertController:(UIAlertController *)alertController
{
    self.alertDefaultActionCount = 0;
    self.alertCancelActionCount = 0;
    self.alertDestroyActionCount = 0;
    
    UIAlertAction *actionWithoutHandler = [UIAlertAction actionWithTitle:@"No Handler"
                                                                   style:UIAlertActionStyleDefault
                                                                 handler:nil];
    UIAlertAction *defaultAction = [UIAlertAction actionWithTitle:@"Default"
                                                            style:UIAlertActionStyleDefault
                                                          handler:^(UIAlertAction *action) {
                                                              ++self.alertDefaultActionCount;
                                                          }];
    UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"Cancel"
                                                           style:UIAlertActionStyleCancel
                                                         handler:^(UIAlertAction *action) {
                                                             ++self.alertCancelActionCount;
                                                         }];
    UIAlertAction *destroyAction = [UIAlertAction actionWithTitle:@"Destroy"
                                                            style:UIAlertActionStyleDestructive
                                                          handler:^(UIAlertAction *action) {
                                                              ++self.alertDestroyActionCount;
                                                          }];
    [alertController addAction:actionWithoutHandler];
    [alertController addAction:defaultAction];
    [alertController addAction:cancelAction];
    [alertController addAction:destroyAction];
    alertController.preferredAction = defaultAction;
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(nullable id)sender
{
    [super prepareForSegue:segue sender:sender];
    if ([segue.identifier isEqualToString:@"presentModal"])
    {
        StoryboardNextViewController *nextVC = segue.destinationViewController;
        nextVC.backgroundColor = UIColor.greenColor;
        nextVC.hideToolbar = NO;
    }
    else if ([segue.identifier isEqualToString:@"show"])
    {
        StoryboardNextViewController *nextVC = segue.destinationViewController;
        nextVC.backgroundColor = UIColor.redColor;
        nextVC.hideToolbar = YES;
    }
}

- (IBAction)showModal
{
    UIViewController *nextVC = [[CodeNextViewController alloc] initWithBackgroundColor:UIColor.purpleColor];
    [self presentViewController:nextVC animated:YES completion:self.viewControllerPresentedCompletion];
}

- (void)presentNonAlert
{
    [self presentViewController:[[UIViewController alloc] init] animated:NO completion:nil];
}

@end

File: SwiftSample/SampleApp/ViewController.swift
import UIKit

final class ViewController: UIViewController {
    @IBOutlet private(set) var showAlertButton: UIButton!
    @IBOutlet private(set) var showActionSheetButton: UIButton!
    @IBOutlet private(set) var seguePresentModalButton: UIButton!
    @IBOutlet private(set) var segueShowButton: UIButton!
    @IBOutlet private(set) var codePresentModalButton: UIButton!

    var alertDefaultActionCount = 0
    var alertCancelActionCount = 0
    var alertDestroyActionCount = 0
    var alertPresentedCompletion: (() -> Void)? = nil
    var viewControllerPresentedCompletion: (() -> Void)? = nil

    @IBAction private func showAlert() {
        let alertController = UIAlertController(title: "Title", message: "Message", preferredStyle: .alert)
        setUpActions(for: alertController)
        alertController.addTextField { textField in
            textField.placeholder = "Placeholder"
        }
        self.present(alertController, animated: true, completion: alertPresentedCompletion)
    }

    @IBAction private func showActionSheet(sender: AnyObject) {
        let alertController = UIAlertController(title: "Title", message: "Message", preferredStyle: .actionSheet)
        setUpActions(for: alertController)

        let popover = alertController.popoverPresentationController
        if let popover = popover {
            popover.sourceView = sender as? UIView
            popover.sourceRect = sender.bounds
            popover.permittedArrowDirections = .any
        }

        self.present(alertController, animated: true)
    }

    private func setUpActions(for alertController: UIAlertController) {
        alertDefaultActionCount = 0
        alertCancelActionCount = 0
        alertDestroyActionCount = 0

        let actionWithoutHandler = UIAlertAction.init(title: "No Handler", style: .default)
        let defaultAction = UIAlertAction.init(title: "Default", style: .default) { _ in
            self.alertDefaultActionCount += 1
        }
        let cancelAction = UIAlertAction.init(title: "Cancel", style: .cancel) { _ in
            self.alertCancelActionCount += 1
        }
        let destroyAction = UIAlertAction.init(title: "Destroy", style: .destructive) { _ in
            self.alertDestroyActionCount += 1
        }

        alertController.addAction(actionWithoutHandler)
        alertController.addAction(defaultAction)
        alertController.addAction(cancelAction)
        alertController.addAction(destroyAction)
        alertController.preferredAction = defaultAction
    }

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        super.prepare(for: segue, sender: sender)
        switch segue.identifier {
        case "presentModal"?:
            guard let nextVC = segue.destination as? StoryboardNextViewController else { return }
            nextVC.backgroundColor = .green
            nextVC.hideToolbar = false
        case "show"?:
            guard let nextVC = segue.destination as? StoryboardNextViewController else { return }
            nextVC.backgroundColor = .red
            nextVC.hideToolbar = true
        default:
            return
        }
    }

    @IBAction private func showModal() {
        let nextVC = CodeNextViewController(backgroundColor: .purple)
        self.present(nextVC, animated: true, completion: viewControllerPresentedCompletion)
    }
}
